### ISSUE TITLE
You cannot call a method on a null-valued expression

### ISSUE THREAD

**Issue by Teazane** [rmbolger/Posh-ACME#185]
Hello! As you proposed here: https://github.com/rmbolger/Posh-ACME/issues/146#issuecomment-541884652, here is a new issue for 3.8.0 version of Posh-ACME. It already happend on 3.7.0 version, but it is still here after update.

As explained, I get this message:
```
Invoke-ACME : You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\3.8.0\Private\Update-PAOrder.ps1:55 char:29
+ ...      $response = Invoke-ACME $header ([String]::Empty) $acct -EA Stop
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Invoke-ACME], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull,Invoke-ACME
```

I use Posh-ACME functions in a custom script. It begins to set PA server and account (with Set-PAServer and New-PAAccount/Set-PAAccount) and then creates an order with New-PAOrder. That's the moment I get the error. I then use "Get-PAOrder | Get-PAAuthorizations" command but the error occurs just a step before at New-PAOrder. 

It worked when I tested it for the first time, and now I get this. What did I do wrong?

**Comment by Teazane** [comment]
Maybe you would like to have more details about my code. Here it is:

```
# Set LE server and LE account
Function SetLEServerAndAccount{
	$ConstructedCertPath = ""
	# Set the correct server
	Set-PAServer LE_PROD 

	# Check if LE account already exist on the given certificate path
	$accountDirectoryList = dir $ConstructedCertPath -Directory | Select-Object name
	# If no account is existing at CertPath location
	if ($accountDirectoryList -eq $null -or $accountDirectoryList.Count -eq 0) {
		# LE account creation with terms of service acceptation.
		$account = New-PAAccount -AcceptTOS -Contact $EmailContact
	# If accounts are present at CertPath location, choose (or create) the correct one
	} else {
		$adminAccountContact = "mailto:"+$EmailContact
		$notExistingAdminAccount = $true
		foreach($accountId in $accountDirectoryList.name){
			$account = GET-PAAccount $accountId
			if ($account.contact -eq $adminAccountContact){
				Set-PAAccount $accountId
				$notExistingAdminAccount = $false
				break
			}
		}
		if($notExistingAdminAccount){
			# LE account creation with terms of service acceptation.
			$account = New-PAAccount -AcceptTOS -Contact $EmailContact
		}
	}
}

# Complete procedure to get new certificates
# Doc: https://github.com/rmbolger/Posh-ACME/wiki/%28Advanced%29-Semi-Manual-DNS-Challenge-Validation
Function GetNewCertificate {
	# Prepare the certificate order giving domain and PFX password.
	New-PAOrder $CertificateDomains -PfxPass $PFXKey  # THE ERROR IS RAISED HERE
        [...]
```

**Comment by rmbolger** [comment]
Apologies for the delay on this, @Teazane. I've been working on another issue. Thanks for the report though. The full code will definitely help track this down once I have some time to look at it.

**Comment by rmbolger** [comment]
So it looks like your code isn't doing anything terribly complicated. It's a `Set-PAServer` call followed by either a `New-PAAccount` or `Set-PAAccount` call and then a `New-PAOrder` call. When I try to follow the same basic flow on my machine I get no errors. So I'm wondering if something in the account object go screwy in a previous version.

Does this consistently happen for a particular account or for all accounts? Does it happen on new accounts created since you upgraded to 3.8?

As a side note, you can probably simplify the code that checks for an existing account by utilizing `Get-PAAccount -List`. Then you don't need to worry about running your own `dir` against a constructed filesystem path. I might do something like this:

```powershell
$account = Get-PAAccount -List | Where-Object { $_.contact -eq "mailto:"+$EmailContact } | Select -First 1
if (-not $account) {
    $account = New-PAAccount -AcceptTOS -Contact $EmailContact
} else {
    Set-PAAccount $account.id
}
```

**Comment by rmbolger** [comment]
There's definitely something strange going on. I recall an issue in the past where other modules or .NET related stuff installed has tried to extend native .NET types like String to include additional stuff that ends up messing with the JSON conversion process and ultimately breaking things in weird ways.

Can you try this from a different machine?

**Comment by Teazane** [comment]
Hello! Sorry for the delay, I had to investigate different things on the machine.

I've tried from a different account on the same machine and the problem just desappeared. It seems that the user I was using before has Internet connection restrictions and cannot contact Let's Encrypt server. The connection was not allowed and so no response was sent. Posh-ACME was probably waiting for a parameter contained in this response and crashed. 

In conclusion: this bug has nothing to do with Posh-ACME which works perfectly xD ! If you want to improve your module, based on this error, the only thing you can do is improving your error message because the null-value was caused by the absence of Internet connection (which is finally a very common and precise problem). 

Thank your very much for your help and reactivity!

**Comment by rmbolger** [comment]
Thanks for the follow up and suggestion!
